spec/jstests-spec.js

```diff
@@ -0,0 +1,185 @@
+import  {initializeTippy, generateGraph, getUrl } from "../src/main/webapp/script.js";
+import cytoscape from "cytoscape";
+
+describe("Checking that fetch url is correctly constructed", function() {
+  let numLayers = {};
+  beforeEach(function () {
+    numLayers = document.createElement("input");
+    numLayers.id = "num-layers";
+  })
+  afterEach(function () {
+    document.body.innerHTML = '';
+  })
+  it("Input valid large value for depth", function() {
```

Comment:
I think the convention for the `it` descriptions is to lowercase the first word, and have it form a complete sentence with "it". Some examples for your tests could be
`it("accepts valid large value")`
`it("uses default value 3 if no user input")`
`it("rounds decimal")`
I vaguely remember that this helps with the error messages? You can double check that by writing a failing test and running it to see if the error message makes more sense this way :)

Reply:
Oh that makes so much more sense. Thanks!